src/crystal/system/unix/path.cr

```diff
@@ -10,7 +10,7 @@ module Crystal::System::Path
       pwd = uninitialized LibC::Passwd
       pwd_pointer = pointerof(pwd)
       ret = nil
-      System.retry_with_buffer("getpwuid_r", User::GETPW_R_SIZE_MAX) do |buf|
+      System.retry_with_buffer("getpwuid_r", GETPW_R_SIZE_MAX) do |buf|
```

Comment:
I think we could refactor this to use the implementation of `System.getpwuid` as well. But you can leave that for a later refactor if you want.

Reply:
I pushed a patch to refactor this. It slightly changes the error in the case (if it can happen?) of not being able to find the current user for `.path` . It's still a runtime error, but the message changes from `getpwuid_r: Undefined error: 0 (RuntimeError)` to `Could not determine current user's home directory` which is probably better?